basetest.pm

```diff
@@ -715,6 +731,51 @@ sub parse_serial_output_qemu {
     return;
 }
 
+sub search_for_expected_autoinst_failures {
+    my ($self) = @_;
+    # autoinst failures defined in distri (test can override them)
+    my $failures = $self->{autoinst_failures};
+
+    my $die = 0;
+    my %regexp_matched;
+    # loop line by line
+    open(my $AUTOINSTLOG, "<", "autoinst-log.txt") || die('Could not open autoinst-log.txt for reading');
+    seek($AUTOINSTLOG, $autoinst_log_pos, 0) || die('Could not seek within autoinst-log.txt');
+    while (my $line = <$AUTOINSTLOG>) {
+        chomp $line;
+        for my $regexp_table (@{$failures}) {
+            my $regexp  = $regexp_table->{pattern};
+            my $message = $regexp_table->{message};
+            my $type    = $regexp_table->{type};
+
+            # Input parameters validation
+            die "Wrong type defined for serial failure. Only 'soft' or 'hard' allowed. Got: $type" if $type !~ /^soft|hard|fatal$/;
+            die "Message not defined for serial failure for the pattern: '$regexp', type: $type" if !defined $message;
+
+            # If you want to match a simple string please be sure that you create it with quotemeta
+            if (!exists $regexp_matched{$regexp} and $line =~ /$regexp/) {
+                $regexp_matched{$regexp} = 1;
+                my $fail_type = 'softfail';
+                if ($type =~ 'hard|fatal') {
+                    $die                   = 1;
+                    $fail_type             = 'fail';
+                    $self->{fatal_failure} = $type eq 'fatal';
+                }
+                $self->record_resultfile($message, $message . " - autoinst-log error: $line", result => $fail_type);
+                $self->{result} = $fail_type;
+            }
+        }
+    }
+    $fp_pos = tell $AUTOINSTLOG;
```

Comment:
Global symbol "$fp_pos" requires explicit package name at ../basetest.pm line 769.

Reply:
I hate perl - should be fixed now.